File: 1.또또의 모험/프로젝트_문서/6_자동화_스크립트/send_slack.mjs
import { WebClient } from '@slack/web-api';
import { SLACK_TOKEN, CHANNEL_ID } from './API_KEY/config.mjs';

// 토큰 확인
if (!SLACK_TOKEN || SLACK_TOKEN === "xoxb-your-slack-bot-token-here") {
    console.error("❌ Error: SLACK_TOKEN이 설정되지 않았습니다. API_KEY/config.mjs 파일을 확인해주세요.");
    process.exit(1);
}

const client = new WebClient(SLACK_TOKEN);

async function sendSlackMessage(channel, text) {
    try {
        const result = await client.chat.postMessage({
            channel: channel,
            text: text,
        });
        console.log(`✅ Message sent successfully: ${result.ts}`);
    } catch (error) {
        console.error(`❌ Error sending message: ${error.message}`);
    }
}

// 실행
(async () => {
    const message = "헬로 월드! 안티그래비티에서 보냄";
    console.log(`Sending message to ${CHANNEL_ID}...`);
    await sendSlackMessage(CHANNEL_ID, message);
})();

File: 1.또또의 모험/프로젝트_문서/6_자동화_스크립트/send_slack.py
import sys
import os
from slack_sdk import WebClient
from slack_sdk.errors import SlackApiError

# config.py가 있는 경로 추가
current_dir = os.path.dirname(os.path.abspath(__file__))
config_path = os.path.join(current_dir, 'API_KEY')
sys.path.append(config_path)

try:
    from config import SLACK_TOKEN
except ImportError:
    print("Error: 'config.py' not found in API_KEY directory or SLACK_TOKEN is missing.")
    sys.exit(1)

def send_slack_message(token, channel, text):
    client = WebClient(token=token)
    try:
        response = client.chat_postMessage(
            channel=channel,
            text=text
        )
        print(f"Message sent successfully: {response['ts']}")
    except SlackApiError as e:
        print(f"Error sending message: {e.response['error']}")

if __name__ == "__main__":
    # 기본 채널 설정 (config.py에 CHANNEL_ID가 없으면 '#general' 사용)
    try:
        from config import CHANNEL_ID
        channel = CHANNEL_ID
    except ImportError:
        channel = "#general"

    message = "헬로 월드! 안티그래비티에서 보냄"
    
    print(f"Sending message to {channel}...")
    send_slack_message(SLACK_TOKEN, channel, message)
